apps/judicial-system/api/src/app/modules/file/file.service.ts

```diff
@@ -33,7 +34,9 @@ export class FileService {
     headers.set('cookie', req.headers.cookie as string)
 
     const result = await fetch(
-      `${this.config.backendUrl}/api/case/${id}/${route}`,
+      `${this.config.backendUrl}/api/case/${id}/${route}${
+        forceRegeneration ? '?forceRegeneration=true' : ''
```

Comment:
Having this as a query parameter seems a bit iffy to me 🤔 Being able to control whether or not the pdf is re-generated or not by changing the URL seems a bit weird. What do you think @gudjong, @Arnarkari93?

Reply:
I think it is fine. We could flip the logic and have a parameter `useSigned` with default true on the backend. So by setting that to false the server would not look for a signed version of the ruling.

The issue is not really about generating or not generating. It is about getting the signed ruling or not.